### ISSUE TITLE
Automatic airplane mode?

### ISSUE THREAD

**Issue by trdischat** [airgap-it/airgap-vault#134]
Not sure if this is possible, but could we add an option to Airgap Vault to automatically put the phone into airplane mode when the app is launched?  Even better would be an option to force airplane mode to stay enabled whenever the app is running.

**Comment by AndreasGassmann** [comment]
This is an interesting idea and we have also thought about it.

I am not sure what functionality the APIs of Android and iOS support. Making sure airplane mode is set when the app is launched should be possible, but I'm not sure if the phone can be forced to stay in airplane mode.

For Android, we are currently investigating what options exist here and how to lock devices down to reduce the attack vector. I'll leave this issue open for future reference.

**Comment by SiNONiMiTY** [comment]
> This is an interesting idea and we have also thought about it.
> 
> I am not sure what functionality the APIs of Android and iOS support. Making sure airplane mode is set when the app is launched should be possible, but I'm not sure if the phone can be forced to stay in airplane mode.
> 
> For Android, we are currently investigating what options exist here and how to lock devices down to reduce the attack vector. I'll leave this issue open for future reference.

How about a helper Accessibility Service? This is used by other apps to automate tasks. For example, cleaner apps use an accessibility service to clear cache of other apps.

**Comment by recolic** [comment]
It's not a good idea to force anything.

What if I just want to use app for some testing? What if I already destroyed wireless driver from linux kernel? What if I'm implementing my own firewall solution?

It's a bad idea to force people to stay secured in your standard. That's what big company like Google/Amazon does to protect their interest. Don't learn from them.

**Comment by martin-braun** [comment]
I agree with @recolic. You are meant to install this app on an offline device. If you don't do it, you have to care for your security on your own. Imho, stock Android should already implement turning off network permissions for individual apps, like GrapheneOS does. That's how things should be handled.

If you are worried about other apps leaking your clipboard or something, well if that is possible, they can also just store your clipboard in local storage and upload it when internet access is back. Not sure this is really a threat, but I don't understand the purpose of this feature request then.

Imho, this issue should be closed.
